### PROBLEMA 1 - Modelo de regresion lineal

Para estudiar la relación existente entre el precio y el número de
habitaciones de una vivienda en Madrid disponemos de los datos referidos a 300
operaciones de venta, de los que se deduce que el precio medio es 250.000€, el
coeficiente de variación del precio es 0’25, el número medio de habitaciones es de 3, el
coeficiente de variación para esta última variable es 0’10 y finalmente el coeficiente de
correlación entre ambas variables se sitúa en 0’8.

    Ajusta un modelo lineal entre las dos variables

#### Solución

Hay una relación existente entre el precio y el numero de habitaciones, en la cual la varibale independiente es X como el numero de habitaciones e Y como el precio, por lo tanto es una recta de regresión de Y sobre X.

#### Datos:

$ n : $ Tamño de la muestra 300

$ \bar{y} : $ Precio medio 250 000

$ CV_{\bar{y}} : $ Coeficente variacion del precio medio 0.25

$ \bar{x} : $ Número medio de habitaciones 3

$ CV_{\bar{x}} :$ Coeficiente de variación del numero de habitaciones 0.10

$ r_{XY} = \frac{S_{XY}}{S_{X}*S_{Y}}$ Coeficente de correlación entre ambas varibales 0.8

In [2]:
import math

In [14]:
n = 300
mu_y = 250000
cv_y = 0.25
mu_x = 3
cv_x = 0.10
rxy = 0.8

#### De los coeficones de variacion hallamos las desviaciones estandar (típicas) 


_Sabemos que:__

$ CV_{z} = \frac{\sigma_{Z}}{\mu_{Z}} $

_Entonces:_

_Para y:_

$ CV_{y} = \frac{\sigma_{y}}{\mu_{y}} $

$ CV_{y}*\mu{_y} = \sigma_{y} $


_Para x :_

$ CV_{x} = \frac{\sigma_{x}}{\mu_{x}} $

$ CV_{x}*\mu_{x} = \sigma_{x} $

#### Sustituyendo 

In [18]:
sy_sigma = cv_y*mu_y
sy_sigma

62500.0

In [20]:
sx_sigma = cv_x*mu_x
sx_sigma

0.30000000000000004

#### Dado el modelo 

#### $Y = \beta_{0} +\beta_{1}*X $

__Tenemos :__

$ \beta_{1} = r_{xy} * \frac{\sigma_{y}}{\sigma_{x}}$

_Donde  $ \sigma_{z} : $ es la desviacion estandar de z_

$ \beta_{0} = \bar{Y} - \beta_{1}*\bar{X} $

In [48]:
b1 = rxy*(sy_sigma/sx_sigma)
b1

166666.66666666666

In [49]:
b0 = mu_y - (b1*mu_x)
b0

-250000.0

#### Por lo tanto el modelo queda como:

In [50]:
print("Y = {} + {}*X".format(b0,b1))

Y = -250000.0 + 166666.66666666666*X


Como la pendiente es positiva, el precio aumenta según aumenta el número de
habitaciones

__Determina cuánto debería pedirse, de acuerdo con el mercado, por una vivienda de
dos habitaciones.__

In [57]:
def predict(b0,b1,x):
    print("Y = {} + {}*{}".format(b0,b1,x))
    return (b0 + (b1*x))

In [62]:
print(predict(b0,b1,2))

Y = -250000.0 + 166666.66666666666*2
83333.33333333331


#### Valoracion del grado de ajuste

__Sabemos que__

$$ R^2 = (r_{XY})^2 = (\frac{S_{XY}}{S_{X}S_{Y}})^2$$

Entonces 

In [70]:
print("{0:.0f} % de fiabilidad del modelo ".format(int((rxy**2)*100)))

64 % de fiabilidad del modelo 


explica el 64% de la variabilidad de la variable Y (Precio) en
función de X (número de habitaciones). No es un porcentaje muy alto (cercano a
100%) pero es aceptable.